File: code_samples/r/algorithms_dynamic_programming.r
# Algorithms: Dynamic Programming
# AI/ML Training Sample

DynamicProgramming <- setRefClass(
    "DynamicProgramming",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- DynamicProgramming$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")

File: code_samples/lua/algorithms_dynamic_programming.lua
-- Algorithms: Dynamic Programming
-- AI/ML Training Sample

DynamicProgramming = {}
DynamicProgramming.__ index = DynamicProgramming

function DynamicProgramming:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function DynamicProgramming:process(input)
    self.data = input
end

function DynamicProgramming:getData()
    return self.data
end

function DynamicProgramming:validate()
    return #self.data > 0
end

-- Example usage
local instance = DynamicProgramming:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))
